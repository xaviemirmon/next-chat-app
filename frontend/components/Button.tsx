import styles from "./Button.module.css";

export const Button = ({
  onClick,
  children,
  variant = "primary",
}: {
  onClick?: () => void;
  children: React.ReactNode;
  variant?: string;
}) => (
  <button className={styles[variant]} onClick={onClick}>
    {children}
  </button>
);
