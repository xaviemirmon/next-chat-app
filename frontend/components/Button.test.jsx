import { render } from "@testing-library/react";

import { Button } from "./Button";

describe("Primary button", () => {
  it("Matches snapshot without icon", () => {
    const { asFragment } = render(<Button>Test</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
  it("Matches snapshot with icon", () => {
    const { asFragment } = render(<Button variant="icon">Button</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
