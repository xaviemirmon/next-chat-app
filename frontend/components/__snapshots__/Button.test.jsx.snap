// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary button Matches snapshot with icon 1`] = `
<DocumentFragment>
  <button
    class="icon"
  >
    Button
  </button>
</DocumentFragment>
`;

exports[`Primary button Matches snapshot without icon 1`] = `
<DocumentFragment>
  <button
    class="primary"
  >
    Test
  </button>
</DocumentFragment>
`;
